zerver/lib/upload.py

```diff
@@ -85,6 +85,16 @@
 # through a sanitization function.
 
 
+# https://github.com/boto/botocore/issues/2644 means that the IMDS
+# request _always_ pulls from the environment.  Monkey-patch the
+# `should_bypass_proxies` function if we need to skip them, based
+# on S3_SKIP_PROXY.
+if settings.S3_SKIP_PROXY is True or (
+    settings.S3_SKIP_PROXY is None and not settings.S3_KEY
```

Comment:
Why isn't this just `if settings.S3_SKIP_PROXY is True`?

Reply:
It's a nullable field, where None means "generally do the right thing."  Which is "skip the proxy" only if you don't have secret keys set, since that's the case you need the role metadata.

Writing this out now, though, I think there's a reasonable argument for just defaulting this to True, since the corner case of "you have something S3-compatible that requires your custom proxy" is probably pretty small.

Reply:
Yeah, that seems correct to me; might require a bit of documentation in our smokescreen/proxy configuration notes.

Reply:
Added a short doc section.

It's also possible we should just drop the flag, and we can add one when it's necessary.  That almost makes the documentation harder, though, since it's just "oh, and one more thing"